api/src/main/java/org/openmrs/module/pihcore/PihCoreActivator.java

```diff
@@ -220,6 +222,48 @@ public void setGlobalProperties(Config config) {
         }
     }
 
+    public User setupCommCareUser(){
+
+	    User user = null;
+        String commCareUserName = Context.getRuntimeProperties().getProperty("commcare.username", "commCareUser");
+        String commCareUserPassword = Context.getRuntimeProperties().getProperty("commcare.password", "CommCare12");
+        if (StringUtils.isNotBlank(commCareUserName)) {
+            user = Context.getUserService().getUserByUsername(commCareUserName);
+            if (user == null) {
+                // create a new user if one does not already exist
+                user = new User();
+            }
+            user.setUsername(commCareUserName);
+            Person person = user.getPerson();
+            if (person == null) {
+                person = new Person();
+                person.addName(new PersonName("CommCare", "", "Integration"));
+                person.setGender("M");
+                user.setPerson(person);
+            }
+
+            user.addRole(Context.getUserService().getRole("Authenticated"));
+            user.addRole(Context.getUserService().getRole("Privilege Level: Full"));
+            if (user.getId() == null ) {
+                // create a new record
+                user = Context.getUserService().createUser(user, commCareUserPassword);
+            } else {
+                // update an existing record
+                user = Context.getUserService().saveUser(user);
+            }
+            person = user.getPerson();
+            Collection<Provider> providers = Context.getProviderService().getProvidersByPerson(person);
+            if (providers == null || providers.size() < 1 ){
+                // if the user does not have a Provider yet then create one
+                Provider provider = new Provider();
```

Comment:
I *believe* that we will need to hard-code the uuid of this provider in the CommCare integration (so we would need assign a hardcoded uuid here?).